Implement Priority components

Mohammed-qmr opened the pull request:
This PR adds two reusable UI components:

PrioritySelector: Allows users to select task priority with animated color transitions and support for light/dark theme.

PriorityBadge: Displays the selected priority with appropriate icon and color, adapting to the current system theme.

Mohammed-qmr committed:
Implement PrioritySelector and PriorityBadge with dynamic theming

devfalah commented on app/src/main/java/com/qudus/tudee/data/model/Priority.kt:
```diff
@@ -0,0 +1,11 @@
+package com.qudus.tudee.data.model
+import com.qudus.tudee.R
+enum class Priority(
```
Please move it to the UI package and rename it to PriorityUiState

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PriorityBadge.kt:
```diff
@@ -0,0 +1,65 @@
+    MaterialTheme {
+        PriorityBadge(priority = Priority.HIGH, modifier = Modifier.padding(8.dp))
+    }
+}
```
Use @PreviewLightDark instead of duplicating previews

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PriorityBadge.kt:
```diff
@@ -0,0 +1,65 @@
+            Icon(
+                painter = painterResource(priority.iconRes),
+                contentDescription = null,
+                tint = Color.White,
```
use colors from theme

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PriorityBadge.kt:
```diff
@@ -0,0 +1,65 @@
+                tint = Color.White,
+                modifier = Modifier.size(14.dp)
+            )
+            Spacer(modifier = Modifier.width(4.dp))
```
please use padding modifier instead of adding `Spacer`

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PrioritySelector.kt:
```diff
@@ -0,0 +1,129 @@
+}
+
+
+@Preview(showBackground = true, name = "Priority Selector - Light")
+@Composable
+fun PreviewPrioritySelectorLight() {
+    var selectedPriority by remember { mutableStateOf(Priority.MEDIUM) }
+
+    MaterialTheme {
+        PrioritySelector(
+            selectedPriority = selectedPriority,
+            onPrioritySelected = { selectedPriority = it },
+            modifier = Modifier.padding(16.dp)
+        )
+    }
+}
+
+@Preview(showBackground = true, uiMode = android.content.res.Configuration.UI_MODE_NIGHT_YES, name = "Priority Selector - Dark")
+@Composable
+fun PreviewPrioritySelectorDark() {
+    var selectedPriority by remember { mutableStateOf(Priority.MEDIUM) }
+
+    MaterialTheme {
+        PrioritySelector(
+            selectedPriority = selectedPriority,
+            onPrioritySelected = { selectedPriority = it },
+            modifier = Modifier.padding(16.dp)
+        )
+    }
+}
```
Use @PreviewLightDark instead of duplicating previews

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PrioritySelector.kt:
```diff
@@ -0,0 +1,129 @@
+}
+
+@Composable
+private fun PriorityItem(
```
What is the difference between PriorityItem and PriorityBadge?

Mohammed-qmr replied:
The difference is that PriorityBadge is for display only (non-interactive), while PrioritySelector is interactive and allows selection with animations.
To reduce code duplication, you can create a shared reusable component that accepts parameters like shape, color, and click behavior, so both can use the same base with simple customization.

devfalah commented on app/src/main/java/com/qudus/tudee/designSystem/component/PrioritySelector.kt:
```diff
@@ -0,0 +1,129 @@
+        Priority.entries.forEach { priority ->
+            val isSelected = priority == selectedPriority
+            val targetColor = if (isSelected) priority.getColor() else Color(0xFFE0E0E0)
+            val contentColor = if (isSelected) Color.White else Color.Gray
```
please use colors from theme

devfalah commented on app/src/main/java/com/qudus/tudee/ui/util/PriorityExtensions.kt:
```diff
@@ -0,0 +1,18 @@
+import com.qudus.tudee.designSystem.color.darkThemeColor
+import com.qudus.tudee.designSystem.color.lightThemeColor
+
+@Composable
+fun Priority.getColor(): Color {
+    val isDark = isSystemInDarkTheme()
+
+    return when (this) {
+        Priority.HIGH -> if (isDark) darkThemeColor.pinkAccent else lightThemeColor.pinkAccent
+        Priority.MEDIUM -> if (isDark) darkThemeColor.yellowAccent else lightThemeColor.yellowAccent
+        Priority.LOW -> if (isDark) darkThemeColor.greenAccent else lightThemeColor.greenAccent
+    }
+}
```
```suggestion
@Composable
fun Priority.getColor(): Color {
    return when (this) {
        Priority.HIGH -> Theme.colors.pinkAccent
        Priority.MEDIUM -> Theme.colors.yellowAccent
        Priority.LOW -> Theme.colors.greenAccent
    }
}
```

ElhamFadel replied:
Here if you can seprate mapping colors and get color function 
this function not responsible

devfalah commented on app/src/main/res/drawable/alert.xml:
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Rename file to icon_alert

devfalah commented on app/src/main/res/drawable/flag.xml:
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Rename file to icon_flag

devfalah commented on app/src/main/res/drawable/trade_down.xml:
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Rename file to icon_trade_down

devfalah commented on gradle/libs.versions.toml:
```diff
@@ -24,9 +28,13 @@ androidx-ui-tooling-preview = { group = "androidx.compose.ui", name = "ui-toolin
 kotlin-compose = { id = "org.jetbrains.kotlin.plugin.compose", version.ref = "kotlin" }
-
+hilt = { id = "com.google.dagger.hilt.android", version.ref = "hilt" }
+kotlin-kapt = { id = "org.jetbrains.kotlin.kapt", version = "2.0.21" }
```
remove hilt please

devfalah reviewed (commented):
Well done @Mohammed-qmr

skaik-mo commented:
Please follow the naming convention for icons and images as agreed on Slack.
Examples:

    icon_name for icons
    image_name for images

Mohammed-qmr committed:
Refactor priority UI components and cleanup

- Move priority-related logic to ui package and rename enum to PriorityUiState
- Replace hardcoded colors with values from Theme
- Replace Spacer with Modifier.padding where appropriate
- Use @PreviewLightDark to combine light/dark previews
- Rename icons to icon_alert, icon_flag, and icon_trade_down for consistency
- Remove unused Hilt dependencies

Mohammed-qmr committed:
Resolve merge conflict in strings.xml and libs.versions.toml

Mohammed-qmr committed:
git commit -m "Resolved merge conflicts in strings.xml and libs.versions.toml (second)"

Mohammed-qmr committed:
Fix: resolved merge conflict in strings.xml

Mohammed-qmr committed:
Merge branch 'develop' into features/shared-components/priority-component
